File: src/main/typescript/wcardinal/controller/data/internal/s-queue-patch.ts
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

export interface SQueuePatch<V> {
	add_( value: V | null ): void;
	addAll_( values: ArrayLike<V | null> ): void;
	remove_(): void;
	capacity_( capacity: number ): void;
	getWeight_(): number;
	isReset_(): boolean;
	serialize_( result: unknown[] ): void;
}

File: src/main/java/org/wcardinal/controller/data/internal/SQueuePatch.java
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

package org.wcardinal.controller.data.internal;

import java.util.Collection;
import java.util.List;

public interface SQueuePatch<V> extends SPatch {
	void add( V value );
	void addAll( Collection<? extends V> values );
	void remove();
	void capacity( int capacity );
	void apply( SQueueValues<V> queue, List<V> padded, List<V> premoved );
}
